toastedmarshmallow/jit.py

```diff
@@ -278,94 +363,200 @@ def generate_marshall_method_body(schema, on_field, context):
-            inliner = None
-            for field_type, inliner_class in iteritems(INLINERS):
-                if isinstance(field_obj, field_type):
-                    inliner = inliner_class.inline(field_obj, context)
-                    if inliner:
-                        break
+            inliner = inliner_for_field(context, field_obj)
 
             if inliner:
-                value_key = 'value'
-                if not jit_options.get('no_callable_fields'):
-                    assignment_template += 'value = {0}; '.format(
-                        inliner.format(value_key))
-                else:
-                    assignment_template += 'value = {0}; '
-                    value_key = inliner.format('value')
-                assignment_template += 'res["{key}"] = {value_key}'.format(
-                    key=key, value_key=value_key)
+                assignment_template += _generate_inlined_access_template(
+                    inliner, result_key, no_callable_fields)
 
             else:
-                assignment_template += (
-                    'res["{key}"] = {field_symbol}__serialize('
-                    '{value_key}, "{field_name}", obj)'.format(
-                        key=key, field_symbol=field_symbol,
-                        field_name=field_name, value_key=value_key))
+                assignment_template += _generate_fallback_access_template(
+                    context, field_name, field_obj, result_key, value_key)
             if not field_obj._CHECK_ATTRIBUTE:
                 # fields like 'Method' expect to have `None` passed in when
                 # invoking their _serialize method.
                 body += assignment_template.format('None')
-            elif _VALID_IDENTIFIER.match(attr_name):
-                body += on_field.serialize(attr_name, field_symbol,
-                                           assignment_template, field_obj)
+                context.namespace['__marshmallow_missing'] = missing
+                body += 'if res["{key}"] is __marshmallow_missing:'.format(
+                    key=result_key)
+                with body.indent():
+                    body += 'del res["{key}"]'.format(key=result_key)
+
             else:
-                # If attr_name is not a valid python identifier, it can only
-                # be accessed via key lookups.
-                body += DictSerializer().serialize(
+                serializer = on_field
+                if not _VALID_IDENTIFIER.match(attr_name):
+                    # If attr_name is not a valid python identifier, it can only
+                    # be accessed via key lookups.
+                    serializer = DictSerializer(context)
+
+                body += serializer.serialize(
                     attr_name, field_symbol, assignment_template, field_obj)
+
+                if not context.is_serializing and field_obj.load_from:
+                    # Marshmallow has a somewhat counter intuitive behavior.
```

Comment:
TIL.

Is there a test that checks this against the marshmallow behavior? It looks like `test_generate_unmarshall_method_bodies_with_load_from` is testing the generated code, but if marshmallow ever changes its behavior you might not notice.

Reply:
Running the Marshmallow tests themselves would catch that behavior change.